tests/test_selection/test_drop_constant_features.py

```diff
@@ -139,6 +140,32 @@ def test_drop_constant_features_with_list_of_variables(df_constant_features):
     pd.testing.assert_frame_equal(X, df)
 
 
+def test_confirm_variables_argument(df_constant_features):
+    """Test the confirm_variable argument."""
+    # Test confirm_variables set to True allows to deal with elements of
+    # variables that are not in the dataframe to fit.
+    variables = ["Name", "City", "Age", "Marks", "dob", "const_feat_num", "no_existing"]
```

Comment:
cCn we expand this test to confirm the variables when a user passes a list, or a string, and what would happen when variables is None and confirm_variables=True?

And then expand in all other selector tests :p

Thank you!